pkg/util/resources.go

```diff
@@ -39,8 +39,8 @@ func ParseAPIGroupsString(apiGroups string) sets.Set[string] {
 	return groupsSet
 }
 
-// IsResourceGroupAllowed checks if a API group is allowed
-// an empty or nil allowedResources slice is equivalent to allow all,
+// IsResourceGroupAllowed checks if a API group is explicitly allowed by user,
+// an empty or nil allowedResources slice is equivalent to allow all.
```

Comment:
can you please re-enable  `allowedResourceGroups.Insert(apiextensions.GroupName)` as part of this PR? it was disabled exactly because of the throttling issue.

Reply:
Done in https://github.com/kubestellar/kubestellar/pull/1828/commits/215a066fa16f731a4cfc6595239a5834cba6b0f3.

Reply:
Reverted because re-enabling calls for some unit test cases to be updated. Will look at the test cases tmr.

Reply:
`controller_test.go` is junk, for the reason not clearly stated in https://github.com/kubestellar/kubestellar/pull/1775#issuecomment-1951422945 . The fakes do not support the sort of discovery that this controller uses, so the test can not exercise much. We should either delete this junk test or change it to use better fakes or change the controller to use the other kind of discovery.

Reply:
@pdettori : can you please elaborate on what you are asking to be re-enabled?

Reply:
Now I see what @pdettori is talking about.

Reply:
Redid in https://github.com/kubestellar/kubestellar/pull/1828/commits/185029311709dc9ececfac041a272c8009dd5c2d together updated unit tests.